backend/src/main/java/com/javahelp/backend/domain/user/register/client/ClientRegisterInteractor.java

```diff
@@ -0,0 +1,81 @@
+package com.javahelp.backend.domain.user.register.client;
+
+import com.javahelp.backend.data.IUserStore;
+import com.javahelp.backend.domain.user.register.RegisterUserResponse;
+import com.javahelp.backend.domain.user.register.UserRegisterInteractor;
+import com.javahelp.model.user.ClientUserInfo;
+import com.javahelp.model.user.User;
+import com.javahelp.model.user.UserPassword;
+
+/**
+ * An interactor that creates new {@link User} and {@link UserPassword} for this client
+ */
+public class ClientRegisterInteractor extends
+        UserRegisterInteractor<IClientRegisterInputBoundary, ClientUserInfo> {
+
+    IUserStore userStore;
+
+    /**
+     * Creates a new {@link ClientRegisterInteractor} instance
+     *
+     * @param userStore  {@link IUserStore} to use
+     */
+    public ClientRegisterInteractor(IUserStore userStore) {
+        super(userStore);
+    }
+
+    /**
+     * To create a new {@link RegisterUserResponse} entity for this provider based on their info:
+     * @param boundary an instance of {@link IClientRegisterInputBoundary} to get info from the
+     *                 register View Model.
+     *
+     * @return a new {@link RegisterUserResponse} entity that comprises of {@link User},
+     * {@link UserPassword} if success
+     * OR of a {@link String} error message if failed.
+     */
+    @Override
+    public RegisterUserResponse register(IClientRegisterInputBoundary boundary){
+        RegisterUserResponse result = isValidated(boundary);
+
+        if (result != null){
+            // Early return for failure, this contains all info about failure reason
+            return result;
+        }
+
+        // After all checks passed, time to make a new client
+        User user = registerClient(boundary);
+
+        UserPassword userPassword = new UserPassword(boundary.getSaltandHash());
+        User newUser = userStore.create(user, userPassword);
+
+        return new RegisterUserResponse(newUser, userPassword);
```

Comment:
This implementation has nothing to do with clients specifically. It works for very kind of `User` and should just be in the superclass.